libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/Encryptor.swift

```diff
@@ -0,0 +1,180 @@
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+//  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import Foundation
+import CryptoKit
+import SalesforceSDKCommon
+
+
+@objc(SFSDKEncryptor)
+public class Encryptor: NSObject {
+    
+    enum EncryptorError: Error {
+        case combinedBoxFailed
+    }
+
+    // MARK: Symmetric Encrypt/Decrypt
+    @objc @available(swift, obsoleted: 1.0) // Objective-c only wrapper
+    public static func encrypt(data: Data, using key: Data) throws -> Data {
+        let symmetricKey = SymmetricKey(data: key)
+        return try encrypt(data: data, using: symmetricKey)
+    }
+
+    public static func encrypt(data: Data, using key: SymmetricKey) throws -> Data {
+        let sealedBox = try AES.GCM.seal(data, using: key, nonce: AES.GCM.Nonce())
+        if let combined = sealedBox.combined {
+            return combined
+        } else {
+            throw EncryptorError.combinedBoxFailed
+        }
+    }
+    
+    @objc @available(swift, obsoleted: 1.0) // Objective-c only wrapper
+    public static func decrypt(data: Data, using key: Data) throws -> Data {
+        let symmetricKey = SymmetricKey(data: key)
+        return try decrypt(data: data, using: symmetricKey)
+    }
+
+    public static func decrypt(data: Data, using key: SymmetricKey) throws -> Data {
+        let sealedBox = try AES.GCM.SealedBox(combined: data)
+        return try AES.GCM.open(sealedBox, using: key)
+    }
+    
+    // MARK: EC Encrypt/Decrypt
+    static func encrypt(data: Data, privateKey: Data) throws -> Data {
+        let key = try derivedKey(from: privateKey)
+        return try encrypt(data: data, using: key)
+    }
+    
+    static func decrypt(data: Data, privateKey: Data) throws -> Data {
+        let key = try derivedKey(from: privateKey)
+        return try decrypt(data: data, using: key)
+    }
+    
+    static func derivedKey(from privateKey: Data) throws -> SymmetricKey {
+        var sharedSecret: SharedSecret
+        if SecureEnclave.isAvailable && !UIDevice.current.isSimulator() {
+            let key = try SecureEnclave.P256.KeyAgreement.PrivateKey(dataRepresentation: privateKey)
+            sharedSecret = try key.sharedSecretFromKeyAgreement(with: key.publicKey)
+        } else {
+            let key = try P256.KeyAgreement.PrivateKey(rawRepresentation: privateKey)
+            sharedSecret = try key.sharedSecretFromKeyAgreement(with: key.publicKey)
+        }
+        return sharedSecret.x963DerivedSymmetricKey(using: SHA256.self, sharedInfo: Data(), outputByteCount: 32)
```

Comment:
Because the key is being derived from a single key pair, I didn't think it mattered what the sharedInfo is and used `Data()`